### PR TITLE
fix(vite-node): fix source map of inlined node_modules

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

Related
- https://github.com/vitest-dev/vitest/issues/5605
- https://github.com/vitest-dev/vitest/pull/6041
- https://github.com/vitest-dev/vitest/pull/7095
- https://github.com/vitejs/vite/pull/19004
- https://github.com/vitejs/vite/pull/18983
- https://github.com/vitest-dev/vitest/pull/7192
- https://github.com/oxc-project/oxc/issues/9169

Currently ssr transform tries to preserve lines by best effort https://github.com/vitejs/vite/pull/19004, but this is not guaranteed since user plugin can also change lines and also there are some not-so-edge cases ssr transform has't handle it yet (I added it as a test case). 

Considering Vite SSR / module runner is also injecting source map (regardless of user code or inlined deps), I think we should allow this for Vitest / vite-node as well.

Also as an important context, preserving lines for transform is hard for the ecosystem for the following aspects:
- https://github.com/oxc-project/oxc/issues/9169
  - with rolldown/oxc, we might do more transform on plain js not only ssr, but it becomes harder to preserve lines.
- https://github.com/vitejs/vite/pull/18983
  - when try to preserve lines for ssr transform, it'll end up many code squashed into a single line, but that tends to break coverage tooling.
  
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: test/cli/deps/error/index.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import 'node:path'
+
+export function testStack() {
```
Comment (hi-ogawa) [inline test/cli/deps/error/index.js:3]: Currently ssr transform doesn't preserve lines with `export`, so this repro fails on main.

File: packages/vite-node/src/server.ts [hunk 2]
```diff
@@ -428,7 +428,7 @@ export class ViteNodeServer {
     }
 
     const sourcemap = this.options.sourcemap ?? 'inline'
-    if (sourcemap === 'inline' && result && !id.includes('node_modules')) {
+    if (sourcemap === 'inline' && result) {
```
Comment (AriPerkkio) [inline packages/vite-node/src/server.ts:431]: I think similar change was proposed before:
- https://github.com/vitest-dev/vitest/pull/6041#discussion_r1668253799

> We don't inline source map because it's expensive to do for every file in node modules

Comment (hi-ogawa) [reply packages/vite-node/src/server.ts:431]: Yes, we can make it to opt-in if broken debugger and stacktrace for inlined deps are acceptable by default, but I'm leaning towards just aligning with Vite SSR.

Comment (sheremet-va) [reply packages/vite-node/src/server.ts:431]: Yes, we had a discussion with the rolldown team not so long ago and decided to remove this check

**Comment by hi-ogawa** [comment]
New test is not running on CI due to https://github.com/vitest-dev/vitest/pull/7558. We need to merge that first.

**Review by sheremet-va — COMMENTED** [review]
What happens if file already has a `sourcermapUrl` that points to a sourcemap file. What ends up running?

**Comment by hi-ogawa** [comment]
> What happens if file already has a sourcermapUrl that points to a sourcemap file. What ends up running?

Vite's transform pipeline starts with `extractSourcemapFromFile` and it will strip all `sourceMappingURL`, so Vitest won't see `sourceMappingURL` comment on our end. I'll add a test case to verify the correctness of collapsed sourcemap from node_modules.

- extractSourcemapFromFile: https://github.com/vitejs/vite/blob/2c51565ec044904a080ef5649034c37f02212c7b/packages/vite/src/node/server/transformRequest.ts#L296-L300
- mapFileCommentRegex:
https://github.com/vitejs/vite/blob/2c51565ec044904a080ef5649034c37f02212c7b/packages/vite/src/node/server/sourcemap.ts#L163

File: test/cli/test/__snapshots__/stacktraces.test.ts.snap [hunk 3]
```diff
@@ -22,6 +22,132 @@ Error: Something truly horrible has happened!
...
+ ❯ testStack (NODE_MODULES)/@vitest/test-dep-error/transpiled-inline.ts:12:3
+ ❯ error-in-package.test.js:20:28
+     18| 
+     19| test('transpiled inline', () => {
+     20|   testStackTranspiledInline()
+       |                            ^
+     21| })
+     22| 
+
+⎯⎯[4/4]⎯
```
Comment (hi-ogawa) [inline test/cli/test/__snapshots__/stacktraces.test.ts.snap:146]: Error frames don't show the original ts file content because it's not found by `getModuleById`.

https://github.com/vitest-dev/vitest/blob/bbfcf9feeeb412d5668e52d5d34acfc9802ad1d0/packages/vitest/src/node/error.ts#L127-L135

After removing `getModuleById` check, frame gets shown properly, but this probably doesn't matter since published package wouldn't ship `.ts` file, so `existsSync` would fail anyways for actual case.

```
 FAIL  stacktraces/error-in-package.test.js > transpiled
Error: __TEST_STACK_TRANSPILED__
 ❯ innerTestStack ../../../node_modules/.pnpm/@vitest+test-dep-error@file+test+cli+deps+error/node_modules/@vitest/test-dep-error/transpiled.ts:22:9
     20|  */
     21| function innerTestStack() {
     22|   throw new Error('__TEST_STACK_TRANSPILED__')
       |         ^
     23| }
     24| 
 ❯ testStack ../../../node_modules/.pnpm/@vitest+test-dep-error@file+test+cli+deps+error/node_modules/@vitest/test-dep-error/transpiled.ts:12:3
 ❯ stacktraces/error-in-package.test.js:16:22

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/4]⎯

 FAIL  stacktraces/error-in-package.test.js > transpiled inline
Error: __TEST_STACK_TRANSPILED_INLINE__
 ❯ innerTestStack ../../../node_modules/.pnpm/@vitest+test-dep-error@file+test+cli+deps+error/node_modules/@vitest/test-dep-error/transpiled-inline.ts:22:9
     20|  */
     21| function innerTestStack() {
     22|   throw new Error('__TEST_STACK_TRANSPILED_INLINE__')
       |         ^
     23| }
     24| 
 ❯ testStack ../../../node_modules/.pnpm/@vitest+test-dep-error@file+test+cli+deps+error/node_modules/@vitest/test-dep-error/transpiled-inline.ts:12:3
 ❯ stacktraces/error-in-package.test.js:20:28
```
